src/averageAges.js

```diff
@@ -54,9 +84,31 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
+
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
-}
+  const children = people.filter(person => onlyWithSon
+    ? person.sex === 'm'
+    && hasMother(people, person.mother)
+    : hasMother(people, person.mother)
+  );
+
+  const motherAges = children
+    .map(person => person.born - bornMother(people, person.mother));
+
+  const motherAverageAge = calculateAverageAge(motherAges);
+
+  return motherAverageAge;
+};
+
+function hasMother(people, motherName) {
+  return people.some(person => person.name === motherName);
+};
+
+function bornMother(people, motherName) {
+  const mother = people.find(person => person.name === motherName);
+
+  return mother.born;
+};
```

Comment:
You can use find method instead of some, then you'll be able to use only one function here